### PR TITLE
update gdbm to work with the newer version of gdbm-sys

### PR DESCRIPTION
**Pull request by miquels**
After merging this, you'll have to fix up https://github.com/cholcombe973/gdbm/commit/42976c635c637764385788ab6a00100fbd47b5e9 . to not point at my github repo of course.

### CODE REVIEW

**Comment by cholcombe973** [comment]
Thanks for the contribution! I think this looks good. Have you tried running valgrind over it? C bindings always cause me to introduce subtle bugs and I can't tell if this is completely correct just by looking at it.

**Comment by miquels** [comment]
No, but I can do that. You mean running it over the test in test/test.rs, or do youhave something more substantial in mind?

**Comment by cholcombe973** [comment]
Yeah i think running it over test.rs would be fine. If it's clean it should be good to go :)

**Comment by miquels** [comment]
I ran the test under valgrind and it did not give any errors. Well, two warnings about `statx`, but I happened to look at the stat/statx implementation in the standard library, and that's simply the stdlib probing to see if there's statx support. It uses a NULL buffer for that, then checks if it get ENOSYS (system call not supported) or "anything else" (system call supported).

```
$ ls -l target/debug/deps/test-fd9873c3349180f5
-rwxrwxr-x 1 mikevs mikevs 5631608 Jan 20 20:43 target/debug/deps/test-fd9873c3349180f5*
$ valgrind target/debug/deps/test-fd9873c3349180f5
==751552== Memcheck, a memory error detector
==751552== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==751552== Using Valgrind-3.15.0 and LibVEX; rerun with -h for copyright info
==751552== Command: target/debug/deps/test-fd9873c3349180f5
==751552==
==751552== Syscall param statx(file_name) points to unaddressable byte(s)
==751552==    at 0x49B589D: syscall (syscall.S:38)
==751552==    by 0x1A227B: statx (weak.rs:95)
==751552==    by 0x1A227B: std::sys::unix::fs::try_statx (fs.rs:123)
==751552==    by 0x19D517: std::sys::unix::fs::stat (fs.rs:1022)
==751552==    by 0x16B9EC: metadata<&std::path::PathBuf> (fs.rs:1567)
==751552==    by 0x16B9EC: term::terminfo::searcher::get_dbpath_for_term (searcher.rs:50)
==751552==    by 0x16B226: term::terminfo::TermInfo::from_name (mod.rs:83)
==751552==    by 0x16B02B: term::terminfo::TermInfo::from_env (mod.rs:69)
==751552==    by 0x1744D4: new<std::io::stdio::Stdout> (mod.rs:219)
==751552==    by 0x1744D4: term::stdout (lib.rs:62)
==751552==    by 0x14F7F9: test::console::run_tests_console (console.rs:252)
==751552==    by 0x15D976: test::test_main (lib.rs:121)
==751552==    by 0x15ED18: test::test_main_static (lib.rs:140)
==751552==    by 0x13CCD5: test::main (in /home/mikevs/devel/rust/gdbm/target/debug/deps/test-fd9873c3349180f5)
==751552==    by 0x13CF3A: core::ops::function::FnOnce::call_once (function.rs:227)
==751552==  Address 0x0 is not stack'd, malloc'd or (recently) free'd
==751552==
==751552== Syscall param statx(buf) points to unaddressable byte(s)
==751552==    at 0x49B589D: syscall (syscall.S:38)
==751552==    by 0x1A227B: statx (weak.rs:95)
==751552==    by 0x1A227B: std::sys::unix::fs::try_statx (fs.rs:123)
==751552==    by 0x19D517: std::sys::unix::fs::stat (fs.rs:1022)
==751552==    by 0x16B9EC: metadata<&std::path::PathBuf> (fs.rs:1567)
==751552==    by 0x16B9EC: term::terminfo::searcher::get_dbpath_for_term (searcher.rs:50)
==751552==    by 0x16B226: term::terminfo::TermInfo::from_name (mod.rs:83)
==751552==    by 0x16B02B: term::terminfo::TermInfo::from_env (mod.rs:69)
==751552==    by 0x1744D4: new<std::io::stdio::Stdout> (mod.rs:219)
==751552==    by 0x1744D4: term::stdout (lib.rs:62)
==751552==    by 0x14F7F9: test::console::run_tests_console (console.rs:252)
==751552==    by 0x15D976: test::test_main (lib.rs:121)
==751552==    by 0x15ED18: test::test_main_static (lib.rs:140)
==751552==    by 0x13CCD5: test::main (in /home/mikevs/devel/rust/gdbm/target/debug/deps/test-fd9873c3349180f5)
==751552==    by 0x13CF3A: core::ops::function::FnOnce::call_once (function.rs:227)
==751552==  Address 0x0 is not stack'd, malloc'd or (recently) free'd
==751552==

running 1 test
test create_test ... ok

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

==751552==
==751552== HEAP SUMMARY:
==751552==     in use at exit: 0 bytes in 0 blocks
==751552==   total heap usage: 733 allocs, 733 frees, 499,671 bytes allocated
==751552==
==751552== All heap blocks were freed -- no leaks are possible
==751552==
==751552== For lists of detected and suppressed errors, rerun with: -s
==751552== ERROR SUMMARY: 2 errors from 2 contexts (suppressed: 0 from 0)
```

**Comment by miquels** [comment]
However, I noticed that I had to fix the test because of the change in `bitflags` so that flags are now in a sub-namespace instead of at the root of the crate.

What would you like to do about it? I can see two solutions:
- try to bring the old at-the-root consts back, maybe just by duplicating them with `use`
- increment the version to `0.2.0`.

I think incrementing the version to 0.2.0 is appropriate, but it's your decision.

**Comment by cholcombe973** [comment]
Your test looks clean! Thanks for doing that.
Yeah I think just incrementing the version to `0.2.0` is fine :)

File: Cargo.toml [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 authors = ["Chris Holcombe <xfactor973@gmail.com>"]
 homepage = "https://github.com/cholcombe973/gdbm"
 description = "Safe wrapper for gdbm-sys bindings"
 license = "MIT"
 
 [dependencies]
-bitflags = "~0.8"
-gdbm-sys = "~0.2"
+bitflags = "~1.2"
+gdbm-sys = { version = "~0.3", git = "https://github.com/miquels/gdbm-sys" }
```
Comment (cholcombe973) [inline Cargo.toml]: Since we merged the change for gdbm-sys do you want to update this now? If not I can do it later after the merge.

Comment (miquels) [reply Cargo.toml]: Oh yes, sure. I'll increment the version to 0.2.0 as well.

**Comment by miquels** [comment]
One more thing. I made Gdbm `Send`, this has UB / security implications if that is wrong. I feel I need to elaborate on why I think that is  correct.

I checked the C source for global variables, did not find them. There is one thread-local variable, the errno, but that is always written and then read on the same thread. The gdbm 'handle' is a pointer to malloc'ed space, so even if that contains self-refential data, it's not going to be moved by the Rust code, it's what in Rust would be called 'pinned'. So I'm quite sure that this is correct.

**Comment by miquels** [comment]
Hey, just checking in too see if there still is something to be done before you can pull this?

**Review by cholcombe973 — APPROVED** [review]
Great work!

**Pull request merged**